src/common/components/messages/ProfileMessages.tsx

```diff
@@ -0,0 +1,385 @@
+                e.target.value = val;
+              }}
+            />
+          )}
+
+          <div className="top-2 right-2">
+            <Button5 loading={messageTransactionInProgress} v="secondary" type="submit">
+              Send
+            </Button5>
+          </div>
+        </form>
+      </div>
+    </div>
+  );
+};
+
+function Contact(props: { selected: boolean; onSelect?: () => void; user: User }): JSX.Element {
+  return (
+    <div
+      id={props.user.address}
+      className={classNames(
+        ' flex cursor-pointer items-center space-x-2 rounded-md  px-2 py-2',
+        props.selected && 'bg-gray-800'
+      )}
+      onClick={props.onSelect}
+    >
+      <ProfilePFP user={props.user} />
+
+      <div className="text-base">
+        {props.user.profile?.handle ?? shortenAddress(props.user.address)}
+      </div>
+    </div>
+  );
+}
+
+const ONE_WEEK_IN_MS = 7 * 24 * 3600 * 1000;
+
+function MessageBlock(props: { myPubkey: string; messageBlock: MessageAccount[] }) {
+  const sender = props.messageBlock[0].sender.toBase58();
+  const iAmSender = sender === props.myPubkey;
+  const msgIsForMe = !iAmSender;
+
+  const ts = props.messageBlock[0].data.ts;
+  const moreThanAWeekAgo = ts && Date.now() - ts.getTime() > ONE_WEEK_IN_MS;
+  const timestamp = ts
+    ? DateTime.fromJSDate(ts).toFormat(
+        moreThanAWeekAgo
+          ? 'ff' // Jun 9, 2022, 6:19 PM
+          : 'cccc t' // "Friday 3:03 PM"
+      )
+    : null;
+
+  const [firstMsg, ...rest] = props.messageBlock;
+  const msgBaseClasses = classNames(
+    'px-6 py-4 max-w-prose rounded-[40px] ',
+    msgIsForMe ? 'bg-gray-800 text-white ml-4' : ' text-black mr-4',
+    iAmSender && 'bg-colorful-gradient bg-repeat-round '
+  );
+
+  const lastMsgIndex = rest.length - 1;
+
+  return (
+    <div className={classNames('flex items-end ', iAmSender ? 'flex-row-reverse' : '')}>
+      <ProfilePFP user={{ address: sender }} />
+      <div
+        className={classNames(
+          'flex flex-col  space-y-2   text-white opacity-90',
+          msgIsForMe ? 'items-start' : 'items-end '
+        )}
+      >
+        {ts && <div className="ml-6 text-xs text-gray-300">{timestamp}</div>}
+        <div
+          className={classNames(
+            msgBaseClasses,
+
+            msgIsForMe ? 'rounded-bl-none' : 'rounded-br-none'
+          )}
+        >
+          {firstMsg.data.body}
+        </div>
+        {rest.map((msg, i) => (
+          <div
+            key={msg.messageId + msg.data.body}
+            className={classNames(
+              msgBaseClasses,
+              i !== lastMsgIndex ? (msgIsForMe ? 'rounded-l-none' : 'rounded-r-none') : null,
+              i === lastMsgIndex ? (msgIsForMe ? 'rounded-tl-none' : 'rounded-tr-none') : null
+            )}
+          >
+            {msg.data.body}
+          </div>
+        ))}
+      </div>
+    </div>
+  );
+}
+
+function createMessageBlocks(messagesInConversation: MessageAccount[]) {
```

Comment:
just a preference, but a clearer name might be `groupMessages`, and return type would make it clear it's not a functional component

Reply:
The lack of a capital first letter also indicates that it's not a functional component, but I'll add a return type now and move it to the top to avoid confusion :)